modules/java/generator/android-21/java/org/opencv/android/JavaCamera2View.java

```diff
@@ -213,8 +211,9 @@ public void onImageAvailable(ImageReader reader) {
                     assert (planes.length == 3);
                     assert (image.getFormat() == mPreviewFormat);
 
-                    JavaCamera2Frame tempFrame = new JavaCamera2Frame(image);
+                    RotatedCameraFrame tempFrame = new RotatedCameraFrame(new JavaCamera2Frame(image), mFrameRotation);
                     deliverAndDrawFrame(tempFrame);
+                    tempFrame.mFrame.release();
```

Comment:
I think it should be moved inside RotatedCameraFrame.release().

Reply:
I thought about this too, but I think it is not good to release an object inside a class if the object is created outside. RotatedCameraFrame constructor takes a pointer to existing JavaCamera2Frame object, and this object may be used in other place.